pkg/causality/internal/node.go

```diff
@@ -105,36 +105,43 @@ func (n *Node) NodeID() int64 {
 }
 
 // DependOn implements interface internal.SlotNode.
-func (n *Node) DependOn(unresolvedDeps map[int64]*Node, resolvedDeps int) {
+func (n *Node) DependOn(dependencyNodes map[int64]*Node, noDependencyKeyCnt int) {
 	resolvedDependees, removedDependees := int32(0), int32(0)
 
 	depend := func(target *Node) {
 		if target == nil {
 			// For a given Node, every dependency corresponds to a target.
 			// If target is nil it means the dependency doesn't conflict
 			// with any other nodes. However it's still necessary to track
-			// it because Node.tryResolve needs to know it.
-			resolvedDependees = stdatomic.AddInt32(&n.resolvedDependees, 1)
+			// it because Node.tryResolve needs to counting the number of
+			// resolved dependencies.
+			resolvedDependees = stdatomic.AddInt32(&n.resolvedDependencies, 1)
 			stdatomic.StoreInt64(&n.resolvedList[resolvedDependees-1], assignedToAny)
-			removedDependees = stdatomic.AddInt32(&n.removedDependees, 1)
+			removedDependees = stdatomic.AddInt32(&n.removedDependencies, 1)
 			return
 		}
 
 		if target.id == n.id {
 			panic("you cannot depend on yourself")
 		}
-		// Lock target and insert `n` into target.dependers.
+
+		// The target node might be removed or modified in other places, for exmaple
+		// after its corresponding transaction has been executed,
 		target.mu.Lock()
 		defer target.mu.Unlock()
 
 		if target.assignedTo != unassigned {
 			// The target has already been assigned to a worker.
-			resolvedDependees = stdatomic.AddInt32(&n.resolvedDependees, 1)
+			// In this case, record the worker ID in `resolvedList`, and this node
+			// probably can be sent to the same worker and executed sequentially.
+			resolvedDependees = stdatomic.AddInt32(&n.resolvedDependencies, 1)
```

Comment:
You can search and replace all `Dependees` in the package.

Reply:
done